view/src/components/DiffViewWrapper.jsx

```diff
@@ -0,0 +1,221 @@
+  findChangeByNewLineNumber,
+  getChangeKey,
+  getCorrespondingOldLineNumber,
+} from 'react-diff-view';
+import refractor from 'refractor';
+
+/** Show whole function if function has max this amount of line. */
+const WHOLE_FUNCTION_SHOW_TRASHOLD = 25;
+/** Amount of lines to show around calling function
+ * if function is too long to show whole. */
+const LINES_OF_CONTEXT = 10;
+
+function isFunctionLong(startLine, endLine) {
+  const lengthOfFunction = endLine - startLine + 1;
+  if (lengthOfFunction > WHOLE_FUNCTION_SHOW_TRASHOLD) {
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Creates file structure necessary for diff visualisation.
+ * Usage - when there is no diff between files, it is needed to create
+ * the file structure by this function.
+ * It creates first hunk containing one line of code (line)
+ * to be able later adds other hunks (lines of code).
+ * @param {String} oldCode - Old text of source code.
+ * @returns Return file structure containing file with hunk.
+ */
+function createFileStructure(oldCode, oldStart, newStart) {
+  const linesOfCode = oldCode.split('\n');
+  const lineOfCode = linesOfCode.slice(oldStart - 1, oldStart);
+  const hunkFromLine = textLinesToHunk(lineOfCode, oldStart, newStart);
+  const file = {
+    type: 'modify',
+    hunks: [hunkFromLine],
+  };
+  return file;
+}
+
+/**
+ * Transform lines of code (to be highlighted) to array which Diff accepts.
+ * @param {Number} linesToShow - Line of code to show.
+ * @returns Returns array of lines to be showed acceptable by Diff.
+ */
+function getLinesToHighlight(linesToShow, hunks) {
+  if (linesToShow) {
+    const oldChange = findChangeByOldLineNumber(hunks, linesToShow[0]);
+    const newChange = findChangeByNewLineNumber(hunks, linesToShow[1]);
+    if (oldChange && newChange) {
+      const oldChangeKey = getChangeKey(oldChange);
+      const newChangeKey = getChangeKey(newChange);
+      if (oldChangeKey === newChangeKey) return [oldChangeKey];
+      return [oldChangeKey, newChangeKey];
+    }
+  }
+  return [];
+}
+
+function getSyntaxHighlightningTokens(hunks) {
+  const options = {
+    highlight: true,
+    refractor,
+    language: 'c',
+  };
+  return tokenize(hunks, options);
+}
+
+/**
+ * Shows code from oldStart to end line with diff hunks.
+ * @param {Object} props
+ * @param {string} props.oldCode - Old source code.
+ * @param {string} props.diff - Output of unified diff of old and new source code.
+ * @param {number} props.oldStart - Start line of old code from which code should be shown.
+ * @param {number} props.end - End line of old code to which code should be shown.
+ * @param {Boolean} [props.showDiff=true] - If it should colored difference in code.
+ * @param {Number} [props.linesToShow=null] - Lines (old, new) of code which
+ *                                            should be shown and highlighted.
+ * @return Return component for visualisation of diff for given code.
+ */
+export default function DiffViewWrapper({
```

Comment:
The logic for choosing which hunks are rendered is not trivial. Maybe a comment somewhere would be helpful.